### PR TITLE
Dev/search basic mechanic

### PR DESCRIPTION
**Pull request by Diaskhan**
Basic Search
- [x] Cancel token  when closing a searchWindow
- [x] double click or enter must navigate to file or folder (if folder not must drill down, must select forlder but not going inside it)

BUGS
- [x] when a new window is showed, it doesnt activated (setfocus), so in general focus still leaved on Mainwindow. 
- [x] fix backspace and enter behaviour when we renamin file or folder inside Datagrid
- [x] when user hit enter, must drill down to the folder, and select first item (.. folder)
- [x] when user hit backspace the focus must be on folder that we was previously selected

### CODE REVIEW

File: src/SmartCommander/Assets/Resources.Designer.cs [hunk 1]
```diff
@@ -626,13 +635,5 @@ public static string Zip {
...
         }
-        /// <summary>
-        ///   AllowOnlyOneInstance
-        /// </summary>
-        public static string AllowOnlyOneInstance {
-            get {
-                return ResourceManager.GetString("AllowOnlyOneInstance", resourceCulture);
-            }
-        }
     }
```
Comment (anovik) [inline src/SmartCommander/Assets/Resources.Designer.cs:638]: It seems the changes in this file can be reverted?

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        get => _isSearching;
+        set => this.RaiseAndSetIfChanged(ref _isSearching, value);
+    }
+
+    public BulkObservableCollection<string> SearchResults { get; }
+    public ReactiveCommand<Unit, Unit> StartSearchCommand { get; }
+    public ReactiveCommand<Unit, Unit> CancelSearchCommand { get; }
+
+    public FileSearchViewModel(string folder = "")
+    {
```
Comment (anovik) [inline src/SmartCommander/ViewModels/FileSearchViewModel.cs:47]: Could you please address the warnings?

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            foreach (var subDir in subDirectories)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await SearchAsync(subDir, searchPattern, cancellationToken);
+            }
+        }
+
+        catch (OperationCanceledException)
+        {
+            Console.WriteLine("Поиск был отменен.");
```
Comment (anovik) [inline src/SmartCommander/ViewModels/FileSearchViewModel.cs:84]: Is it debug message?

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        _cancellationTokenSource = new CancellationTokenSource();
+        _timer = new Timer(OnTimerTick, null, 0, 500);
+        await Task.Run(() => SearchAsync(CurrentFolder, FileMask, _cancellationTokenSource.Token));
+
+        IsSearching = false;
+    }
+
+    private void OnTimerTick(object state)
+    {
+
```
Comment (anovik) [inline src/SmartCommander/ViewModels/FileSearchViewModel.cs:114]: extra line, please remove

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        CancelSearchCommand = ReactiveCommand.Create(CancelSearch);
+    }
+
+    public async Task<bool> SearchAsync(string folderPath, string searchPattern, CancellationToken cancellationToken)
+    {
+        try
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+
```
Comment (anovik) [inline src/SmartCommander/ViewModels/FileSearchViewModel.cs:61]: extra line, please remove

**Review by anovik — COMMENTED** [review]
Looks promising, could you please address my comments?

File: src/SmartCommander/Assets/Resources.Designer.cs (hunk 1 above)
Comment (Diaskhan) [reply src/SmartCommander/Assets/Resources.Designer.cs:638]: its auto-sorting of generated files, aplies automatically in VS

**Comment by Diaskhan** [comment]
sorry, its draft 
@anovik  could u Add some requiremnts for this FileSearch Window ?

**Comment by anovik** [comment]
No problem. I think we can keep it simple in the first iteration. Options for file search can be: search in file names (later can add regex and search in file content), search either in current dir only or in all subdirectories.

In the first iteration, we can display file list in the window with possibility of click and navigating to it/its folder.

Later we can add moving all found files to panel.

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs (hunk 4 above)
Comment (Diaskhan) [reply src/SmartCommander/ViewModels/FileSearchViewModel.cs:114]: fixed

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs (hunk 5 above)
Comment (Diaskhan) [reply src/SmartCommander/ViewModels/FileSearchViewModel.cs:61]: fixed

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs (hunk 3 above)
Comment (Diaskhan) [reply src/SmartCommander/ViewModels/FileSearchViewModel.cs:84]: fixed

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs (hunk 2 above)
Comment (Diaskhan) [reply src/SmartCommander/ViewModels/FileSearchViewModel.cs:47]: fixed

File: src/SmartCommander/Views/MainWindow.axaml.cs [hunk 6]
```diff
@@ -61,6 +62,7 @@ public MainWindow()
         {
             var dialog = new T2();
             dialog.DataContext = interaction.Input;
+            dialog.Activate();
```
Comment (Diaskhan) [inline src/SmartCommander/Views/MainWindow.axaml.cs:65]: fixed, when a new dialog is showed, focus will be on new window

File: src/SmartCommander/Views/FilesPane.axaml.cs [hunk 7]
```diff
@@ -29,7 +36,31 @@ public FilesPane()
...
+            {
+                focusManager = TopLevel.GetTopLevel((Visual)PaneDataGrid)?.FocusManager;
+                PaneDataGrid.AddHandler(KeyDownEvent, dataGrid_PreviewKeyDown, RoutingStrategies.Tunnel);
+                PaneDataGrid.Focus();
+            }
+        }
+
+
+
+        public void dataGrid_PreviewKeyDown(object? sender, KeyEventArgs e)
```
Comment (Diaskhan) [inline src/SmartCommander/Views/FilesPane.axaml.cs:49]: fix backspace and enter behaviour when we renamin file or folder inside Datagrid

File: src/SmartCommander/ViewModels/FileSearchViewModel.cs [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+            foreach (var subDir in subDirectories)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                await SearchAsync(subDir, searchPattern, cancellationToken);
+            }
+        }
+
+        catch (OperationCanceledException)
+        {
+            Console.WriteLine("Search was canceled");
```
Comment (anovik) [inline src/SmartCommander/ViewModels/FileSearchViewModel.cs:81]: Can you comment out the debug messages with Console.Write/WriteLine for now? I used Serilog logger for debugging myself earlier in this project, and I think we need to add it again in a separate PR and use for all the debugging and logging.

File: src/SmartCommander/Views/FilesPane.axaml.cs [hunk 9]
```diff
@@ -29,8 +36,44 @@ public FilesPane()
...
+                PaneDataGrid.AddHandler(KeyDownEvent, dataGrid_PreviewKeyDown, RoutingStrategies.Tunnel);
+                PaneDataGrid.ScrollIntoView(PaneDataGrid.SelectedItem, null);
+                PaneDataGrid.Focus();
+   
+
+
+                var viewModel = (FilesPaneViewModel?)DataContext;
+                viewModel!.ScrollToItemRequested += (item, column) =>
+                {
+
```
Comment (anovik) [inline src/SmartCommander/Views/FilesPane.axaml.cs:51]: extra line, please remove

File: src/SmartCommander/Views/FilesPane.axaml.cs [hunk 10]
```diff
@@ -29,8 +36,44 @@ public FilesPane()
...
             var PaneDataGrid = this.Get<DataGrid>("PaneDataGrid");
-            PaneDataGrid.Focus();
+            if (PaneDataGrid != null)
+            {
+                focusManager = TopLevel.GetTopLevel((Visual)PaneDataGrid)?.FocusManager;
+                PaneDataGrid.AddHandler(KeyDownEvent, dataGrid_PreviewKeyDown, RoutingStrategies.Tunnel);
+                PaneDataGrid.ScrollIntoView(PaneDataGrid.SelectedItem, null);
+                PaneDataGrid.Focus();
+   
+
```
Comment (anovik) [inline src/SmartCommander/Views/FilesPane.axaml.cs:46]: extra lines, please remove

**Review by anovik — COMMENTED** [review]
Looks good, just a couple of minor comments.

File: src/SmartCommander/Views/FilesPane.axaml.cs (hunk 10 above)
Comment (Diaskhan) [reply src/SmartCommander/Views/FilesPane.axaml.cs]: fixed

File: src/SmartCommander/Views/FilesPane.axaml.cs (hunk 9 above)
Comment (Diaskhan) [reply src/SmartCommander/Views/FilesPane.axaml.cs]: fixed

**Review by anovik — APPROVED** [review]

**Pull request merged**